src/ripple/overlay/impl/Manifest.cpp

```diff
@@ -275,6 +276,17 @@ ManifestDisposition
 ManifestCache::applyManifest (
     Manifest m, ValidatorList& unl, beast::Journal journal)
 {
+    /*
+        Move master public key from permanent trusted key list
+        to manifest cache.
+    */
+    if (unl.trusted(m.masterKey))
+    {
+        auto comment = unl.member (m.masterKey);
+        addTrustedKey (m.masterKey, comment.get());
+        unl.removePermanentKey (m.masterKey);
+    }
```

Comment:
Oh, I just spotted a minor optimization: since `trusted` calls `member`, how about
```
{
    auto unlComment = unl.member (m.masterKey);
    if (unlComment)
    {
        addTrustedKey (m.masterKey, *unlComment);
        unl.removePermanentKey (m.masterKey);
    }
}
````